lock-api/src/main/java/com/palantir/lock/ConjureTimestampRangeTimestampSupplier.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock;
+
+import com.palantir.atlasdb.timelock.api.ConjureTimestampRange;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import java.util.function.LongSupplier;
+
+public final class ConjureTimestampRangeTimestampSupplier implements LongSupplier {
```

Comment:
It's really weird that this kind of class doesn't exist. And ideally it would just be a simple

```
LongStream.rangeClosed().stream().iterator() or something
```

Reply:
Do you want me to change it to this later?